ClangSharp.PInvokeGenerator.Test/StructDeclarationTest.cs

```diff
@@ -91,6 +91,58 @@ public partial struct MyNestedStruct
             await ValidateGeneratedBindings(inputContents, expectedOutputContents);
         }
 
+        [Fact]
+        public async Task SkipNoDefinitionTest()
+        {
+            var inputContents = "typedef struct MyStruct MyStruct;";
+
+            var expectedOutputContents = $@"namespace ClangSharp.Test
+{{
+    public partial struct MyStruct
+    {{
+    }}
+}}
+";
+            await ValidateGeneratedBindings(inputContents, expectedOutputContents);
+        }
+
+        [Theory]
+        [InlineData("unsigned char", "byte")]
+        [InlineData("double", "double")]
+        [InlineData("short", "short")]
+        [InlineData("int", "int")]
+        [InlineData("long long", "long")]
+        [InlineData("char", "sbyte")]
+        [InlineData("float", "float")]
+        [InlineData("unsigned short", "ushort")]
+        [InlineData("unsigned int", "uint")]
+        [InlineData("unsigned long long", "ulong")]
+        public async Task SkipNonDefinitionTest(string nativeType, string expectedManagedType)
+        {
+            var inputContents = $@"typedef struct MyStruct MyStruct;
```

Comment:
Here is a case where the definition exists later in the file. This ensures we only generate:
```cs
struct MyStruct
{
    float r;
}
```

rather than instead generating:
```cs
public partial struct MyStruct
{
}

public partial struct MyStruct
{
    float r;
    float g;
    float b;
}
```

Reply:
Out of curiosity .. where did you find this type of declaration?

Reply:
It happens fairly frequently in COM header files, like `dxgi.h`, as an example.

It is generally used to handle cases where you need to accept a method parameter before the type can actually be defined.